### PR TITLE
WAZO-2261: delete recording media

### PR DESCRIPTION
**Pull request by bloom1**
Depends-On: https://github.com/wazo-platform/wazo-call-logd-client/pull/9

### CODE REVIEW

**Comment by bloom1** [comment]
recheck

File: wazo_call_logd/plugins/cdr/http.py [hunk 1]
```diff
@@ -212,3 +212,30 @@ def get(self, cdr_id, recording_uuid):
...
+
+        try:
+            self.service.delete_media(cdr_id, recording_uuid, recording.path)
+            return '', 204
+        except PermissionError:
+            logger.error('Permission denied: "%s"', recording.path)
+            raise RecordingMediaFSPermissionException(recording_uuid, recording.path)
+        except FileNotFoundError:
+            logger.error('Recording file not found: "%s"', recording.path)
+            raise RecordingMediaFSNotFoundException(recording_uuid, recording.path)
```
Comment (fblackburn1) [inline wazo_call_logd/plugins/cdr/http.py:241]: Like other DELETE endpoints, we should return 204 if the resource is already deleted (and be sure that deleted attribute is True)

File: integration_tests/suite/helpers/database.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def wrapped_function(self, *args, **kwargs):
             with self.database.queries() as queries:
                 recording['uuid'] = queries.insert_recording(**recording)
             try:
-                return func(self, recording, *args, **kwargs)
+                return func(self, *args, recording, **kwargs)
```
Comment (fblackburn1) [inline integration_tests/suite/helpers/database.py:100]: :laughing: I see this one too in my branch, I hope it doesn't give you too much headache to find it

**Review by fblackburn1 — CHANGES_REQUESTED** [review]

File: integration_tests/suite/test_cdr_recording.py [hunk 3]
```diff
@@ -148,3 +150,140 @@ def test_get_media_using_token_tenant_query_string(self):
...
+
+    @call_log(**{'id': 1}, recordings=[{'path': None}])
+    def test_delete_media_with_recording_already_deleted(self):
+        rec_uuid = self.call_logd.cdr.get_by_id(1)['recordings'][0]['uuid']
+        assert_that(
+            calling(self.call_logd.cdr.delete_recording_media).with_args(1, rec_uuid),
+            raises(CallLogdError).matching(
+                has_properties(status_code=400, error_id='recording-media-not-found')
+            ),
+        )
```
Comment (fblackburn1) [inline integration_tests/suite/test_cdr_recording.py:220]: I'm not sure if this behavior still make sense with your last modification?
It doesn't return error if your media is not found and deleted is false
But return an error if your media is not found and deleted is true
Maybe we should only return 204, deleted or not and doesn't bother the user with this logic. What do you think?

**Review by fblackburn1 — APPROVED** [review]

**Pull request merged**